File: test-signal.py
import signal, os, time

is_breaked = False

def handler(signum, frame):
	global is_breaked
	is_breaked = True
	print ('handler %d' % signum)

signal.signal(signal.SIGTERM, handler)

while not is_breaked:
    time.sleep(1)

print "exit main"

File: test-signal.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>


int is_breaked = 1;

void my_terminate (int param)
{
  printf ("Terminating program...%d\n", param);
  is_breaked = 0;
}

int main ()
{
  signal (SIGTERM,my_terminate);
 
  while(is_breaked == 1)sleep(1);

  printf("main exited\n");

  return 0;
}
